thoth/storages/graph/postgres.py

```diff
@@ -5001,6 +5001,48 @@ def sync_dependency_monkey_result(self, document: dict) -> None:
 
                 inspection_run.dependency_monkey_run_id = dependency_monkey_run.id
 
+    def get_python_package_required_symbols(
+        self, package_name: str, package_version: str, index_url: str
+    ) -> List[str]:
+        """Get required symbols for a Python package in a specified version."""
+        package_name = self.normalize_python_package_name(package_name)
+        package_version = self.normalize_python_package_version(package_version)
+
+        query = self._session.query(PythonPackageVersion)
+
+        query = (
+            query.filter(PythonPackageVersion.package_name == package_name)
+            .filter(PythonPackageVersion.package_version == package_version)
+            .join((HasArtifact, PythonPackageVersion.python_artifacts))
+            .join((PythonArtifact, HasArtifact.python_artifact))
+            .with_entities(PythonArtifact.versioned_symbols)
+        )
+
+        return query.first()   # If query result is empty returns None
+
+
+    def get_image_symbols(
+        self,
+        environment_name: str,
+        os_name: str,
+        os_version: str,
+        cuda_version: str,
+        python_version: str
+    ) -> List[str]:
+        """Get symbols associated with a given image."""
+        query = (
+            self._session.query(PackageExtractRun)
+            .filter(PackageExtractRun.analysis_document_id == environment_name)
```

Comment:
What is `environment_name` in this case and why you filter that with the `analysis_document_id`? Maybe you mean `environment_type`? BUILDTIME and RUNTIME? In that case you should enforce the use of `EnvironmentTypeEnum`.